tree/dataframe/src/RMetaData.cxx

```diff
@@ -0,0 +1,107 @@
+// Author: Ivan Kabadzhov CERN  10/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOT/RDF/RMetaData.hxx"
+#include <stdexcept> // std::logic_error
+
+#include <iostream>
+
+namespace ROOT {
+namespace RDF {
+namespace Experimental {
+
+void RMetaData::AddMetaData(const std::string &cat, int val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddMetaData(const std::string &cat, double val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddMetaData(const std::string &cat, const std::string &val)
+{
+   fJson[cat] = val;
+}
+
+void RMetaData::AddJSONMetaData(const std::string &cat, const nlohmann::json &val)
+{
+   fJson[cat] = val;
+}
+
+std::string RMetaData::Dump(const std::string &cat) const
+{
+   return fJson[cat].dump();
+}
+
+int RMetaData::GetI(const std::string &cat) const
```

Comment:
We have still not fixed how to handle type mismatches:
* actual string, asking for double --> (i.e. having "langauage":"C++", and then GetD("langauage")) --> error
* actual string, asking for int --> similar to above --> error
* actual double, asking for string --> return the double in a string, e.g. "3.14"
* actual double, asking for int --> ?
* actual int, asking for string --> similar to point 4, e.g. "42"
* actual int, asking for double --> cast the int to double so 14 becomes 14.

Reply:
currently throw directly on type mismatch.